apps/web/src/app/(private)/evaluations/(evaluation)/[evaluationUuid]/editor/_components/Editor/Advanced/index.tsx

```diff
@@ -36,32 +32,37 @@ export default function AdvancedEvaluationEditor({
   providerApiKeys,
   freeRunsCount,
 }: {
-  evaluation: EvaluationDto
+  evaluation: AdvancedEvaluationDto
   defaultPrompt: string
   providerApiKeys?: ProviderApiKey[]
   freeRunsCount?: number
 }) {
   const { findEvaluation, isLoading, update, isUpdating } = useEvaluations({
     fallbackData: [serverEvaluation],
   })
-  const evaluation = findEvaluation(serverEvaluation.uuid)
-  const evaluationMetadata = useMemo(
-    () => evaluation?.metadata as EvaluationMetadataLlmAsJudgeAdvanced,
-    [evaluation],
-  )
+  const evaluation = findEvaluation(
+    serverEvaluation.uuid,
+  ) as AdvancedEvaluationDto
   const { data: providers } = useProviderApiKeys({
     fallbackData: providerApiKeys,
   })
   const [value, setValue] = useState(defaultPrompt)
   const { metadata, runReadMetadata } = useMetadata()
 
+  const [promptlVersion, setPromptlVersion] = useState(
+    evaluation.metadata.promptlVersion,
+  )
+  useEffect(() => {
+    setPromptlVersion(evaluation.metadata.promptlVersion)
+  }, [evaluation.metadata.promptlVersion])
```

Comment:
Why this `useEffect`, why not pass `evaluation.metadata.promptVersion` to `runReadMetadata`?

Reply:
Because the “upgrade” button doesn’t update the evaluation in the backend. Instead, it configures the editor with the new version (the ⁠promptlVersion state). When the user saves the evaluation, it will then save it.

Reply:
Talked offline